### PR TITLE
implemented userdetails & added authorities

### PR DESCRIPTION
**Pull request by oksana-rudenko**

### CODE REVIEW

File: src/main/java/springboot/onlinebookstore/controller/BookController.java [hunk 1]
```diff
@@ -26,27 +27,31 @@
...
 
+    @PreAuthorize("hasRole('ROLE_USER')")
     @GetMapping
     @Operation(summary = "Get all books",
             description = "Getting all available books from book store")
     public List<BookDto> getAll(Pageable pageable) {
         return bookService.findAll(pageable);
     }
 
+    @PreAuthorize("hasRole('ROLE_USER')")
```
Comment (koval-kn) [inline src/main/java/springboot/onlinebookstore/controller/BookController.java]: so admin would not have access to that endpoint?

File: src/main/java/springboot/onlinebookstore/controller/BookController.java [hunk 2]
```diff
@@ -55,13 +60,15 @@ public BookDto update(@PathVariable Long id,
...
 
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
     @DeleteMapping("/{id}")
     @Operation(summary = "Delete a book by its ID",
             description = "Soft deleting a book by its ID from book store")
     public void deleteById(@PathVariable Long id) {
         bookService.deleteById(id);
     }
 
+    @PreAuthorize("hasRole('ROLE_USER')")
```
Comment (koval-kn) [inline src/main/java/springboot/onlinebookstore/controller/BookController.java]: so admin would not have access to that endpoint?

**Review by koval-kn — CHANGES_REQUESTED** [review]

File: src/main/java/springboot/onlinebookstore/config/SecurityConfig.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        return http
+                .cors(AbstractHttpConfigurer::disable)
+                .csrf(AbstractHttpConfigurer::disable)
+                .authorizeHttpRequests(
+                        auth -> auth
+                                .requestMatchers("/auth/**", "/swagger-ui/**")
```
Comment (sokimaaa) [inline src/main/java/springboot/onlinebookstore/config/SecurityConfig.java]: append with "/error" to allow users see error page

File: src/main/java/springboot/onlinebookstore/config/SecurityConfig.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import org.springframework.security.web.SecurityFilterChain;
+
+@EnableMethodSecurity
+@Configuration
+public class SecurityConfig {
+    private final UserDetailsService userDetailsService;
+
+    public SecurityConfig(UserDetailsService userDetailsService) {
+        this.userDetailsService = userDetailsService;
+    }
```
Comment (sokimaaa) [inline src/main/java/springboot/onlinebookstore/config/SecurityConfig.java]: You dont need to inject dependencies in such way when you use @Configuration and @Bean. 

You can directly inject via method params.
```suggestion
```

File: src/main/java/springboot/onlinebookstore/model/Role.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Entity
+@Data
+@SQLDelete(sql = "UPDATE roles SET is_deleted = true WHERE id = ?")
+@Where(clause = "is_deleted = false")
+@Table(name = "roles")
+public class Role {
```
Comment (sokimaaa) [inline src/main/java/springboot/onlinebookstore/model/Role.java:16]: Could the role be deleted or created?

You instantiate all roles in `RoleName`. 
And how your BookController will handle this? e.g. @PreAuthorize("hasRole('ROLE_ADMIN')")

**Review by Sanderfuck — APPROVED** [review]

**Pull request merged**